### PR TITLE
Added /join command

### PR DESCRIPTION
**Pull request by bcomnes**
Also starts a decent framework for more commands.  Almost closes https://github.com/moose-team/friends/issues/94

Still need to handle # removal

### CODE REVIEW

**Comment by bcomnes** [comment]
Seems to work actually without any # processing.

File: app.js [hunk 1]
```diff
@@ -227,6 +227,23 @@ function App (el) {
...
 
+  self.on('executeCommand', function (commandStr) {
+    var words = commandStr.split(' ')
+    var command = words[0].substring(1, words[0].length).toLowerCase()
+    var message
+
+    switch (command) {
+      case 'join':
+        words.shift()
+        var channel = words.join(' ')
```
Comment (beaugunderson) [inline app.js]: not sure what happens if you join a channel with a space, does it work?

File: lib/elements/composer.js [hunk 2]
```diff
@@ -74,8 +74,12 @@ Composer.prototype.autocomplete = function (text, ownUsername) {
...
 Composer.prototype.submit = function (node) {
   var self = this
-  self.send('sendMessage', node.value)
-  node.value = ''
+  if (node.value.charAt(0) === '/') {
+    self.send('executeCommand', node.value)
+  } else {
+    self.send('sendMessage', node.value)
+  }
+  ode.value = ''
```
Comment (Trott) [inline lib/elements/composer.js]: typo here (obviously)

File: app.js (hunk 1 above)
Comment (tunnckoCore) [inline app.js]: you mean `/join `? good q

maybe we need one more event for handle different type of commands. i.e just `/part` command and `/join (#)channel`

Comment (tunnckoCore) [inline app.js]: and `indexOf(' ')` check before `var words = commandStr.split(' ')`

**Comment by bcomnes** [comment]
Do we support channels with spaces?

**Comment by bcomnes** [comment]
I'm fine with part, but /wc is standard IRC too no?

File: app.js (hunk 1 above)
Comment (bcomnes) [inline app.js]: This would just join the channel of the first word.

**Comment by beaugunderson** [comment]
oh /wc is from irssi looks like (`/window close`), i was always BitchX or mIRC :)

feel free to add both

**Comment by bcomnes** [comment]
I gotta run to bed now :[  If this is the general direction you want to take feel free to merge and keep working on it.  I dont want to hold the train up.  

See you all tomorrow!

File: lib/elements/composer.js (hunk 2 above)
Comment (bcomnes) [inline lib/elements/composer.js]: Gah!  Yah `standard` just picked this up

**Comment by pokemonmegaman** [comment]
While we're at it, might as well also add an alias for `/leave` to do the same as `/wc`.

**Comment by Trott** [comment]
Right now, channel names can have spaces or, for that matter, it appears any character. Is that a bug or a feature? It does seem that if you are going to type a channel name, white space would be expected as a delimiter.

**Comment by tunnckoCore** [comment]
> While we're at it, might as well also add an alias for /leave to do the same as /wc.

yea, and I think we first do the `/alias` api and signature, then these things would be easy.

> It does seem that if you are going to type a channel name, white space would be expected as a delimiter.

atm, yes, but its easy for change

**Comment by bcomnes** [comment]
@pokemonmegaman Cool!  `self.data.activeChannel.name`  sounds promising.  I'll get that fixed up now.

**Comment by bcomnes** [comment]
@tunnckoCore  I'm a little on your first coupe comments in the thread.  Mind adding line numbers?

**Comment by bcomnes** [comment]
Part/leave/wc works great now.  @pokemonmegaman++ for pointout out that object structure to me.  I should have just looked a few lines above XD

File: app.js [hunk 3]
```diff
@@ -227,6 +227,27 @@ function App (el) {
...
+        self.emit('addChannel', channel)
+        break
+      case 'wc':
+      case 'part':
+      case 'leave':
+        self.emit('leaveChannel', self.data.activeChannel.name)
+        break
+      default:
+        console.log('Unrecognized command')
+        break
```
Comment (ngoldman) [inline app.js]: @bcomnes this looks great! one thought -- maybe if the command is unrecognized it just passes through as a message instead?

Comment (bcomnes) [inline app.js]: Ah yeah! Accidental democratic voting for new commands lol

**Comment by bcomnes** [comment]
@ngoldman unrecognized commands now just send!  They also log the unrecognized command.

**Comment by ngoldman** [comment]
:+1:

**Comment by tunnckoCore** [comment]
looks great, good job! :+1: :)

**Pull request merged**